src/Service.Tests/Configuration/Telemetry/FileSinkTests.cs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Serilog;
+using Serilog.Core;
+using static Azure.DataApiBuilder.Service.Tests.Configuration.ConfigurationTests;
+
+namespace Azure.DataApiBuilder.Service.Tests.Configuration.Telemetry;
+
+/// <summary>
+/// Contains tests for File Sink functionality.
+/// </summary>
+[TestClass, TestCategory(TestCategory.MSSQL)]
+public class FileSinkTests
+{
+    public TestContext TestContext { get; set; }
+
+    private const string CONFIG_WITH_TELEMETRY = "dab-file-sink-test-config.json";
+    private const string CONFIG_WITHOUT_TELEMETRY = "dab-no-file-sink-test-config.json";
+    private static RuntimeConfig _configuration;
+
+    /// <summary>
+    /// This is a helper function that creates runtime config file with specified telemetry options.
+    /// </summary>
+    /// <param name="configFileName">Name of the config file to be created.</param>
+    /// <param name="isFileSinkEnabled">Whether telemetry is enabled or not.</param>
+    /// <param name="fileSinkPath">Path where logs will be sent to.</param>
+    /// <param name="rollingInterval">Time it takes for logs to roll over to next file.</param>
+    public static void SetUpTelemetryInConfig(string configFileName, bool isFileSinkEnabled, string fileSinkPath, RollingInterval? rollingInterval = null)
```

Comment:
why does this function need to be public? Should be private if its used only within this test